srtcore/atomic.h

```diff
@@ -82,31 +122,120 @@ class atomic {
                            sizeof(T) == 8,
                        "Only types of size 1, 2, 4 or 8 are supported");
 
-  atomic() : value_(static_cast<T>(0)) {}
+  atomic()
+    : value_(static_cast<T>(0))
+  {
+#if defined(ATOMIC_USE_POSIX_MUTEX) && (ATOMIC_USE_POSIX_MUTEX == 1)
+    const int lResult = pthread_mutex_init(&mutex_, NULL);
+    if (lResult != 0)
+    {
+       perror("Initializing Mutex");
+       assert(lResult == 0);
+    }
+#endif
+  }
+
+  explicit atomic(const T value)
+    : value_(value)
+  {
+#if defined(ATOMIC_USE_POSIX_MUTEX) && (ATOMIC_USE_POSIX_MUTEX == 1)
+    const int lResult = pthread_mutex_init(&mutex_, NULL);
+    if (lResult != 0)
+    {
+       perror("Initializing Mutex");
+       assert(lResult == 0);
+    }
+#endif
+  }
+
+  ~atomic()
+  {
+#if defined(ATOMIC_USE_POSIX_MUTEX) && (ATOMIC_USE_POSIX_MUTEX == 1)
+    const int lResult = pthread_mutex_destroy(&mutex_);
+    if (lResult != 0)
+    {
+       perror("Destroying Mutex");
+       assert(lResult == 0);
+    }
+#endif
+  }
 
-  explicit atomic(const T value) : value_(value) {}
+#if defined(ATOMIC_USE_POSIX_MUTEX) && (ATOMIC_USE_POSIX_MUTEX == 1)
+  class PosixMutexLockGuard
```

Comment:
A similar class `srt::sync::ScopedLock`. is already defined in `sync.h`.
`atomic.h` is included from `sync.h`, meaning that its definition is not yet available here.

Reply:
Generally my suggestion is that if now the atomics definitions are dependent on POSIX threads, then there has been a dependency introduced and the declarations should be reshuffled. The order better be then: 1) POSIX wrapper, 2) atomics, and all compiled into one header file - `sync.h`.

Reply:
Submitted a commit that implements `ATOMIC_USE_SRT_SYNC_MUTEX`, replacing `ATOMIC_USE_POSIX_MUTEX` .